### PR TITLE
Fixes issues with HttpClientManager

### PR DESCRIPTION
**Pull request by Bond-009**
Fixes a crash when using the `Post` method with content that isn't a byte array.

### CODE REVIEW

File: Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs [hunk 1]
```diff
@@ -324,125 +302,84 @@ private async Task<HttpResponseInfo> SendAsyncInternal(HttpRequestOptions option
...
+                {
+                    httpWebRequest.Content = new StringContent(options.RequestContent);
+                }
+                else
+                {
+                    httpWebRequest.Content = new ByteArrayContent(Array.Empty<byte>());
                 }
-                catch (Exception ex)
+                /*
+                var contentType = options.RequestContentType ?? "application/x-www-form-urlencoded";
```
Comment (cvium) [inline Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs]: Any reason not to remove this?

File: Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs [hunk 2]
```diff
@@ -324,125 +302,84 @@ private async Task<HttpResponseInfo> SendAsyncInternal(HttpRequestOptions option
...
-                        await stream.CopyToAsync(memoryStream).ConfigureAwait(false);
-                        memoryStream.Position = 0;
+            options.CancellationToken.ThrowIfCancellationRequested();
 
-                        return GetResponseInfo(response, memoryStream, memoryStream.Length, null);
-                    }
-                }
-            }
-            catch (OperationCanceledException ex)
+            /*if (!options.BufferContent)
```
Comment (cvium) [inline Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs]: Same here

File: Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs (hunk 1 above)
Comment (Bond-009) [reply Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs]: Content types aren't being set atm (they are, but not as requested), this is there as a reminder (for now).

File: Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs (hunk 2 above)
Comment (Bond-009) [reply Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs]: This was already there, didn't touch that afaik

File: Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs (hunk 1 above)
Comment (cvium) [reply Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs]: Maybe add a TODO to make it searchable?

File: Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs (hunk 2 above)
Comment (cvium) [reply Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs]: Maybe add a TODO to make it searchable?

**Pull request closed without merging**